[runtime] Complete support for exception marshalling on CoreCLR.

rolfbjarne opened the pull request:
* This also required an empty implementation of the toggle ref machinery,
  because the ObjectiveCMarshal.Initialize call to initialize unhandled
  exception support requires passing toggle ref callbacks as well. The
  complete implementation is more complex, and will come in an uncoming PR.

* The TestConstrainedGenericType test can now be re-enabled, after a few
  updates.

vs-mobiletools-engineering-service2 commented:
### :x: [PR Build] Tests failed on Build :x:

Tests failed on Build. 
* [Azure DevOps](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4812989&view=ms.vss-test-web.test-result-details)
* [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.21/4812989/sim/;/tests/vsdrops_index.html)
# API diff
:white_check_mark: API Diff from stable
<details><summary>View API diff</summary>

* iOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.21/4812989/apidiff/;/ios-api-diff.html) [gist](https://gist.github.com/a037904d79b060b959cd101f2df1cf0c)
* macOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.21/4812989/apidiff/;/mac-api-diff.html) [gist](https://gist.github.com/6e59f701fcf9e09143d68537ae05a7ba)
* macCat [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.21/4812989/apidiff/;/maccat-api-diff.html) [gist](https://gist.github.com/ab6f0b096800afb7144690767b1d2016)
* macCatiOS  [gist](https://gist.github.com/dfec35aaaef965d0210348c2082fadfd)
* tvOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.21/4812989/apidiff/;/tvos-api-diff.html) [gist](https://gist.github.com/c2ef6a0e127e161708090805480f2e21)
* watchOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.21/4812989/apidiff/;/watchos-api-diff.html) [gist](https://gist.github.com/d6bf8576b4385f606652716cc34a3b8b)
</details>

# API & Generator diff

:white_check_mark: [API Diff (from PR only)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.21/4812989/apigeneratordiff/;//api-diff.html) (no change)
:white_check_mark: [Generator Diff](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.21/4812989/apigeneratordiff/;//generator-diff/index.html) (no change)

# GitHub pages

Results can be found in the following github pages (it might take some time to publish):

* [Test results](https://xamarin.github.io/macios.ci/pr/PR11734/4812989/HtmlReport-sim/tests/vsdrops_index.html)
* [API diff ](https://xamarin.github.io/macios.ci/pr/PR11734/4812989/HtmlReport-sim/api-diff/api-diff.html)
* [API & Generator diff](https://xamarin.github.io/macios.ci/pr/PR11734/4812989/apicomparison/api-diff.html)


# Test results

<details>
<summary>2 tests failed, 170 tests passed.</summary>

## Failed tests

 * monotouch-test/Mac [dotnet]/Debug (CoreCLR) [dotnet]: Failed (Test run failed.
Tests run: 2444 Passed: 2355 Inconclusive: 7 Failed: 3 Ignored: 86)
 * monotouch-test/Mac [dotnet]/Debug (CoreCLR, static registrar) [dotnet]: Failed (Test run failed.
Tests run: 2441 Passed: 2355 Inconclusive: 7 Failed: 1 Ignored: 85)
</details>


[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4812989&view=ms.vss-test-web.test-result-details) on Agent XAMBOT-1099.BigSur'
Merge 11df830d6ce48683463fd3d7191d213d45a2e249 into 51c936457d6ce0f7f596179f0fbe6b294c56f6c7

rolfbjarne commented:
Test failures are unrelated:

* monotouch-test/Mac [dotnet]/Debug (CoreCLR) [dotnet]: Fixed in #11732 and #11733.
* monotouch-test/Mac [dotnet]/Debug (CoreCLR, static registrar) [dotnet]: Fixed in #11733 and #11732.
